lib/puppet/type/netapp_qtree.rb

```diff
@@ -8,23 +8,54 @@
   ensurable
 
   newparam(:name) do
-    desc "The qtree name."
     isnamevar
+    desc "The qtree name."
     validate do |value|
-      unless value =~ /^[\w\-]+$/
+      unless (value =~ /^[\w\-\.]+$/) or (value =~ /^\/\w+\/[\w\-\.]+$/)
          raise ArgumentError, "%s is not a valid qtree name." % value
       end
     end
   end
 
   newparam(:volume) do
     desc "The volume to create qtree against."
+    defaultto do
+      if @resource[:name].count('/') == 2
+        @resource[:name].split('/')[1] 
+      else
+        ""
+      end
+    end
```

Comment:
The way to have various parameters default to values from the namevar (often called "composite namevars") would be to use `self.title_patterns` like such: https://github.com/puppetlabs/puppetlabs-iis/blob/83bb37e6/lib/puppet/type/iis_application.rb#L10-L34

The linked example offers two regular expressions. If the title matches a regex then the match groups of the regex will be the default values for the corresponding parameters. Because this is just the defaults, explicitly passing values would override the defaults from the `self.title_patterns`.

For this netapp change, it would look something like this:
```ruby
  def self.title_patterns
    [
      [
        /^([\w\s\.]+)$/,
        [
          [:qtname],
        ]
      ],
      [
        %r{^/(\w+)/([\w\s\.]+)$},
        [
          [:volume],
          [:qtname],
        ]
      ]
    ]
  end
```